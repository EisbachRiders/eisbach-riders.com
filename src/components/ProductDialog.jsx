import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Img from 'gatsby-image'
import SwipeableViews from 'react-swipeable-views'
import { withStyles } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'
import Button from '@material-ui/core/Button'
import Dialog from '@material-ui/core/Dialog'
import DialogActions from '@material-ui/core/DialogActions'
import DialogContent from '@material-ui/core/DialogContent'
import DialogTitle from '@material-ui/core/DialogTitle'
import IconButton from '@material-ui/core/IconButton'
import CloseIcon from '@material-ui/icons/Close'
import List from '@material-ui/core/List'
import ListItem from '@material-ui/core/ListItem'
import ListItemText from '@material-ui/core/ListItemText'

const styles = theme => ({
  more: {
    width: '100%',
  },
  dialogContainer: {
    display: 'flex',
  },
  dialogImg: {
    flexBasis: '60%',
  },
  dialogContent: {
    flexBasis: '40%',
  },
  dialogTitle: {
    textAlign: 'center',
  },
  closeButton: {
    float: 'right',
  },
  name: {
    paddingTop: 15,
    paddingBottom: 10,
    fontWeight: 600,
    textAlign: 'left',
  },
  price: {
    color: theme.palette.primary.main,
    fontWeight: 600,
    display: 'flex',
    justifyContent: 'space-between',
    paddingBottom: 10,
  },
})

class ProductDialog extends Component {
  state = {
    isDialogOpen: false,
  }

  handleDialogOpen = idx => {
    this.setState({
      isDialogOpen: true,
    })
  }

  handleDialogClose = () => {
    this.setState({ isDialogOpen: false })
  }

  render() {
    const { product, classes } = this.props
    const { isDialogOpen } = this.state
    console.log(product)
    return (
      <div className={classes.root}>
        <Button
          variant="contained"
          className={classes.more}
          onClick={this.handleDialogOpen}
        >
          More
        </Button>
        <Dialog
          open={isDialogOpen}
          onClose={this.handleDialogClose}
          maxWidth="md"
        >
          <DialogTitle id="dialog-title" className={classes.dialogTitle}>
            {product.name}
            <IconButton
              key="close"
              aria-label="Close"
              color="inherit"
              onClick={this.handleDialogClose}
              className={classes.closeButton}
            >
              <CloseIcon />
            </IconButton>
          </DialogTitle>
          <DialogContent>
            <div className={classes.dialogContainer}>
              <div className={classes.dialogImg}>
                <SwipeableViews>
                  {product.images.map((img, idx) => (
                    <Img
                      key={`img${idx}`}
                      fluid={img.node.fluid}
                      alt=""
                      className={classes.img}
                    />
                  ))}
                </SwipeableViews>
              </div>
              <div className={classes.dialogContent}>
                <Typography variant="h6" className={classes.price}>
                  {product.price}
                </Typography>
                <List>
                  {product.more.map((item, idx) => (
                    <ListItem>
                      <ListItemText primary={item} />
                    </ListItem>
                  ))}
                </List>
              </div>
            </div>
          </DialogContent>
        </Dialog>
      </div>
    )
  }
}

ProductDialog.propTypes = {
  classes: PropTypes.object.isRequired,
}

export default withStyles(styles)(ProductDialog)
