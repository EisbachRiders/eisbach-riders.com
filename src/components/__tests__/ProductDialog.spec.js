import React from 'react'
import { render } from 'react-testing-library'
import ProductDialog from '../ProductDialog'

describe('ProductDialog', () => {
  it('renders correctly', () => {
    const mockProps = {
      handleDialogOpen: jest.fn,
      handleDialogClose: jest.fn,
      isDialogOpen: true,
      product: {},
      buttonText: 'mockButtonText',
    }
    const component = render(<ProductDialog {...mockProps} />)
    expect(component).toMatchSnapshot()
  })
})
