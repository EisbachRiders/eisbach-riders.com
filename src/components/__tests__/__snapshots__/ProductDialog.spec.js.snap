// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDialog renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="overflow: hidden; padding-right: 0px;"
  >
    <div
      aria-hidden="true"
    >
      <button
        class="MuiButtonBase-root-18 MuiButton-root-1 MuiButton-outlined-6 MuiButton-outlinedSecondary-8 Hook-button-1ecjfgm"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-2"
        >
          mockButtonText
        </span>
        <span
          class="MuiTouchRipple-root-38"
        />
      </button>
    </div>
    <div
      class="MuiModal-root-36 MuiDialog-root-21"
      role="dialog"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root-45"
        style="opacity: 1; -webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container-24 MuiDialog-scrollPaper-22"
        role="document"
        style="opacity: 1; -webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          class="MuiPaper-root-47 MuiPaper-elevation24-73 MuiPaper-rounded-48 MuiDialog-paper-25 Hook-paper-itjtj3 MuiDialog-paperScrollPaper-26 MuiDialog-paperWidthMd-31"
        >
          <div
            class="MuiDialogTitle-root-74 Hook-dialogTitle-1yaitwv"
            id="dialog-title"
          >
            <h6
              class="MuiTypography-root-75 MuiTypography-h6-85"
            >
              <button
                aria-label="close"
                class="MuiButtonBase-root-18 MuiIconButton-root-105 MuiIconButton-colorInherit-106 Hook-closeButton-op4cyh"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label-110"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root-111"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root-38"
                />
              </button>
            </h6>
          </div>
          <div
            class="MuiDialogContent-root-120 Hook-dialogRoot-lfeyyx"
          >
            <div
              class="Hook-dialogContainer-1iekob2"
            >
              <div
                class="Hook-dialogImg-1sccabg"
              >
                <div
                  style="overflow-x: hidden;"
                >
                  <div
                    class="react-swipeable-view-container"
                    style="flex-direction: row; direction: ltr; display: flex;"
                  />
                </div>
                <div
                  class="MuiPaper-root-47 MuiPaper-elevation0-49 MuiMobileStepper-root-121 MuiMobileStepper-positionStatic-124 Hook-mobileStepper-1emojig"
                >
                  <button
                    aria-label="scroll right"
                    class="MuiButtonBase-root-18 MuiButtonBase-disabled-19 MuiIconButton-root-105 MuiIconButton-disabled-109"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label-110"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root-111"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
                        />
                        <path
                          d="M0 0h24v24H0V0z"
                          fill="none"
                        />
                      </svg>
                    </span>
                  </button>
                  <div
                    class="MuiMobileStepper-dots-125"
                  />
                  <button
                    aria-label="scroll left"
                    class="MuiButtonBase-root-18 MuiIconButton-root-105"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label-110"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root-111"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                        />
                        <path
                          d="M0 0h24v24H0V0z"
                          fill="none"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root-38"
                    />
                  </button>
                </div>
              </div>
              <div
                class="Hook-dialogContent-3d7b3e"
              >
                <div
                  class="Hook-priceContainer-vyf8vb"
                >
                  <h6
                    class="MuiTypography-root-75 MuiTypography-h6-85 Hook-price-1yarhbx"
                  >
                    â‚¬undefined
                  </h6>
                  <div />
                </div>
                <ul
                  class="MuiList-root-129 Hook-scroll-vhpypq"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  >
    <button
      class="MuiButtonBase-root-18 MuiButton-root-1 MuiButton-outlined-6 MuiButton-outlinedSecondary-8 Hook-button-1ecjfgm"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label-2"
      >
        mockButtonText
      </span>
      <span
        class="MuiTouchRipple-root-38"
      />
    </button>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
